shapeshift/web
# feat: custom slippage improvements for UI and cowswap swapper

## Description

Some types of swappers (thorchain streaming swaps) dont support the ability to specify the slippage when getting a quote, and we need a way to propagate this to users. Other swappers (0x) only provide slippage protection for certain pairs.

This PR implements a UI change that shows users what quotes are affected.

- Includes custom slippage capability for cowsap
- 0x only provides slippage protection for specific pairs, so this PR addresses that
- Fixes trade execution UI showing the default slippage instead of custom slippage

## Pull Request Type


## Issue (if applicable)



closes #5779

## Risk

- Risk of broken trades
- Risk of incorrect custom slippage for cow swap

## Testing

- Check trades are working as expected
- Check cowswap custom slippage is correct

### Engineering

Carefully check the implementation of custom slippage for cowswap.

### Operations



## Screenshots (if applicable)

Cowswap default slippage value - see matching payload
![image](https://github.com/shapeshift/web/assets/125113430/32980875-fc9d-4aa0-bb48-5b8c006a48aa)

Cowswap custom slippage value - see matching payload
![image](https://github.com/shapeshift/web/assets/125113430/bfc328da-23a2-4f36-b944-869e247ac19d)

Completed cow trade with custom slippage (note bottom right of UI 
![image](https://github.com/shapeshift/web/assets/125113430/c972172f-6cdd-40b0-8d70-b1efe7ac2a7a)

Video showing slippage UI behaviour

https://github.com/shapeshift/web/assets/125113430/fe60fdef-092d-4f0b-adcd-15aa80470e63
